Find a functionally equivalent code:android.view.ViewGroup.dispatchGenericPointerEvent:COMMENT
Method Modifier: protected   
Comment:/**
 * {@inheritDoc}
 */

@@ -2,21 +2,26 @@
     // Send the event to the child under the pointer.
     final int childrenCount = mChildrenCount;
     if (childrenCount != 0) {
-        final View[] children = mChildren;
         final float x = event.getX();
         final float y = event.getY();
-        final boolean customOrder = isChildrenDrawingOrderEnabled();
+        final ArrayList<View> preorderedList = buildOrderedChildList();
+        final boolean customOrder = preorderedList == null && isChildrenDrawingOrderEnabled();
+        final View[] children = mChildren;
         for (int i = childrenCount - 1; i >= 0; i--) {
-            final int childIndex = customOrder ? getChildDrawingOrder(childrenCount, i) : i;
-            final View child = children[childIndex];
+            int childIndex = customOrder ? getChildDrawingOrder(childrenCount, i) : i;
+            final View child = (preorderedList == null) ? children[childIndex] : preorderedList.get(childIndex);
             if (!canViewReceivePointerEvents(child) || !isTransformedTouchPointInView(x, y, child, null)) {
                 continue;
             }
             if (dispatchTransformedGenericPointerEvent(event, child)) {
+                if (preorderedList != null)
+                    preorderedList.clear();
                 return true;
             }
         }
+        if (preorderedList != null)
+            preorderedList.clear();
     }
     // No child handled the event.  Send it to this view group.
     return super.dispatchGenericPointerEvent(event);

